### ISSUE TITLE
synchronous version of TriggerEventT

### ISSUE THREAD

**Issue by lynnard** [reflex-frp/reflex#92]
Looks like there isn't a version of TriggerEventT that uses synchronous triggering (yet). Do you accept pull requests for this?

(context: I work on libs that concern animation / gaming, so having synchronous versions would be nice)

**Comment by ryantrinkle** [comment]
@lynnard Sorry, looks like this got buried!  Can you give me some more detail on what functionality you were looking for?  I'll make sure it gets on the list to add when possible.

**Comment by lynnard** [comment]
:D yes, this has been a while.

So what I originally wanted is to have a generic monad transformer similar to `TriggerEventT` but instead of pushing triggers into a channel, directly call a firing function encapsulated in it (that was what I was refering to by *synchronous triggering*).

Eventually I went ahead and implemented something at my library side for the gaming needs, which can be seen here <https://github.com/lynnard/reflex-cocos2d/blob/07d822a9d4b6cb529437bff6c12be8a845e1fc88/src/Reflex/Cocos2d/Builder/Base.hs#L119> (I actually used a new type class `FastTriggerEvent` that's almost exactly the same as `TriggerEvent` but gives no guarantee about proper sequencing of firings - as in, they are fired on the spot).

So I think this can be safely closed now.
